### PR TITLE
BTV: Add warning for NaN DeepCSV values and inputs

### PR DESCRIPTION
**Pull request by eacoleman**
If implemented, this PR will:
- Add a warning in the DeepFlavourJetTagsProducer which is triggered when one of the DeepCSV inputs is NaN
- Add a warning which is triggered when the DeepCSV output is NaN

Relevant contacts: @mverzett @emilbols @emanueleusai

### CODE REVIEW

File: RecoBTag/Combined/plugins/DeepFlavourJetTagsProducer.cc [hunk 1]
```diff
@@ -220,6 +220,12 @@ DeepFlavourJetTagsProducer::produce(edm::Event& iEvent, const edm::EventSetup& i
 				else {
 					inputs_[var.name] = vars.get(var.id, var.default_value);
 				}
+
+                //warn if the input is nan
```
Comment (perrotta) [inline RecoBTag/Combined/plugins/DeepFlavourJetTagsProducer.cc]: Please indent correctly

File: RecoBTag/Combined/plugins/DeepFlavourJetTagsProducer.cc [hunk 2]
```diff
@@ -229,6 +235,13 @@ DeepFlavourJetTagsProducer::produce(edm::Event& iEvent, const edm::EventSetup& i
 			for(auto entry : toadd_) {
 				nnout[entry.second] += nnout[entry.first];
 			}
+
+            //warn if the input is nan
```
Comment (perrotta) [inline RecoBTag/Combined/plugins/DeepFlavourJetTagsProducer.cc]: Please indent correctly

**Comment by perrotta** [comment]
@eacoleman : does the corresponding pull request in the master exist? If not, please do so

**Comment by slava77** [comment]
how frequent are these warnings?
Only once in ~100 events is acceptable.

**Comment by eacoleman** [comment]
@slava77 the frequency of these warnings is certainly below this threshold.

**Comment by slava77** [comment]
On 6/22/18 12:29 AM, Evan Coleman wrote:
> @slava77 <https://github.com/slava77> the frequency of these warnings is
> certainly below this threshold.
> 

That's very nice to know.
Please quantify, just in case.
Thank you.

**Comment by eacoleman** [comment]
@perrotta for corresponding PR in master please see #23640.

**Comment by eacoleman** [comment]
@mverzett Can you please quantify for @slava77 the approximate rate of nan values in the DeepCSV tagger? 

@slava77 Just to give a sense, in one affected file, we collected 14 jets with warnings from a sample of over 100,000 events. The error is not critical but is being logged as a warning while we investigate the source of the problem. For example, it may be that this is training-related (e.g. nan weights propagating to jets with certain characteristics) and therefore should be checked regularly as the trainings are adapted. So far we have not seen an indication that this problem should prevent the processing of other events, or of other jets within an impacted event, thus we have not thrown an exception.

**Comment by perrotta** [comment]
please test

**Comment by slava77** [comment]
On 6/22/18 6:45 AM, Evan Coleman wrote:
> @mverzett <https://github.com/mverzett> Can you please quantify for
> @slava77 <https://github.com/slava77> the approximate rate of nan values
> in the DeepCSV tagger?
> 
> @slava77 <https://github.com/slava77> Just to give a sense, in one
> affected file, we collected 14 jets with warnings from a sample of over
> 100,000 events. 

I think that this is a good enough quantification.
Thank you.


The error is not critical but is being logged as a
> warning while we investigate the source of the problem. For example, it
> may be that this is training-related (e.g. nan weights propagating to
> jets with certain characteristics) and therefore should be checked
> regularly as the trainings are adapted. So far we have not seen an
> indication that this problem should prevent the processing of other
> events, or of other jets within an impacted event, thus we have not
> thrown an exception.
>

**Comment by perrotta** [comment]
backport of #23640

**Comment by slava77** [comment]
-1

updates are needed before we can proceed with integration

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/23618	0d936db
- "backport" from #23640 is correct (ignoring changes in spaces). This only adds a LogWarning in cases DeepFlavourJetTagsProducer has nan inputs/outputs.
- jenkins tests pass and comparisons with the baseline show no differences
    - notably, in 136.7611 which has a ```ValueError``` in the master branch tests, there are no warnings in 94X. One difference between the two is the more recent training used since 10_2_X.

**Comment by fabiocos** [comment]
+1

**Pull request merged**